node/actors/executor/src/lib.rs

```diff
@@ -56,18 +56,30 @@ pub struct Executor<S> {
 
 impl<S: WriteBlockStore + 'static> Executor<S> {
     /// Creates a new executor with the specified parameters.
-    pub fn new(
+    pub async fn new(
+        ctx: &ctx::Ctx,
         node_config: ExecutorConfig,
         node_key: node::SecretKey,
         storage: Arc<S>,
     ) -> anyhow::Result<Self> {
+        node_config.validate()?;
         anyhow::ensure!(
             node_config.gossip.key == node_key.public(),
             "config.gossip.key = {:?} doesn't match the secret key {:?}",
             node_config.gossip.key,
             node_key
         );
 
+        // While justifications may differ among nodes for an arbitrary block, we assume that
+        // the genesis block has a hardcoded justification.
+        let first_block = storage.first_block(ctx).await.context("first_block")?;
```

Comment:
I'm not entirely sure whether it's valid for a storage to contain blocks before the genesis block. IMO, it makes reasoning about the storage more difficult, so the current check disallows it.